src/main/java/org/vandeseer/easytable/structure/Row.java

```diff
@@ -38,22 +39,20 @@ private Row(final List<AbstractCell> cells) {
         this.cells = cells;
     }
 
-    boolean hasNext() {
-        return next != null;
-    }
-
     public float getHeight() {
         if (table == null) {
             throw new TableNotYetBuiltException();
         }
 
-        final float maxCellHeight = getCells().stream()
-                .filter(cell -> cell.getRowSpan() == 1)
-                .map(AbstractCell::getHeight)
-                .max(naturalOrder())
-                .orElseThrow(RuntimeException::new);
+        if (height == null) {
+            this.height = getCells().stream()
+                    .filter(cell -> cell.getRowSpan() == 1)
+                    .map(AbstractCell::getHeight)
+                    .max(naturalOrder())
+                    .orElseThrow(RuntimeException::new);
+        }
```

Comment:
I was looking into the performance issue as well and saw this, too. You solved it pretty nicely with the `Float` (which can be `null` and models the absence of the value) – in my tests I just quickly introduced an `Optional`, but your solution is way better :+1: :smiley:

Reply:
Wait stop. 'Optional<>` is better. :)
Optionals are built for this case and `null`s kinda dirty. 

I will change that. :+1:

Reply:
Well, the question is whether one wants to use an `Optional` as a class member. SonarLint for instance complains about it. Also see this answer from Brian Goetz himself on stackoverflow: https://stackoverflow.com/questions/26327957/should-java-8-getters-return-optional-type/26328555#26328555

For class members (since that is implementation detail) using a boxed `Float` is sufficient ;)